MediaProcessor/tests/FFmpegControllerTester.cpp

```diff
@@ -0,0 +1,65 @@
+#include <gtest/gtest.h>
+
+#include <filesystem>
+#include <nlohmann/json.hpp>
+
+#include "../src/AudioProcessor.h"
+#include "../src/ConfigManager.h"
+#include "../src/FFmpegController.h"
+#include "TestUtils.h"
+
+namespace fs = std::filesystem;
+namespace MediaProcessor::Tests {
+
+fs::path testMediaPath = TEST_MEDIA_DIR;
+
+class FFmpegControllerTester : public ::testing::Test {
+   protected:
+    fs::path testVideoPath;
+    fs::path testOutputDir;
+    TestUtils::TestConfigFile testConfigFile;
+    FFmpegConfigManager configManager;
+
+    FFmpegControllerTester() : configManager(FFmpegConfigManager()) {}
+
+    void assertFileExists(fs::path path) {
+        ASSERT_TRUE(fs::exists(path)) << path << " not found.";
+    }
+
+    void SetUp() override {
+        fs::path currentPath = fs::current_path();
+
+        testVideoPath = testMediaPath / "test_video.mkv";
+
+        assertFileExists(testVideoPath);
+
+        testOutputDir = currentPath / "test_output";
+        fs::create_directories(testOutputDir);
+
+        testConfigFile.changeConfigOptions("use_thread_cap", true, "max_threads_if_capped", 1);
+    }
+
+    void TearDown() override {
+        fs::remove_all(testOutputDir);
+    }
+};
+
+// ClassName_MethodName_StateUnderTest_ExpectedBehavior gtest std naming convention
+TEST_F(FFmpegControllerTester, ExtractAudio_ExtractsAudioCorrectly) {
+    fs::path outputPath = testOutputDir / "test_output_audio.wav";
+
+    configManager.setOverwrite(true);
+    configManager.setInputFilePath(testVideoPath);
+    configManager.setAudioSampleRate(48000);
+    configManager.setAudioChannels(1);
+    configManager.setAudioCodec(AudioCodec::AAC);
+    configManager.setOutputFilePath(outputPath);
+
+    FFmpegController controller = FFmpegController(configManager);
+
+    EXPECT_EQ(controller.extractAudio(), true);
+
+    EXPECT_TRUE(fs::exists(outputPath));
+}
```

Comment:
Could you take a look at [this utility](https://github.com/omeryusufyagci/fast-music-remover/blob/8c4952ea58d8c195c8ba08ede40252cc7359b67d/MediaProcessor/tests/AudioProcessorTester.cpp#L59) to see if you can reuse it here? It's more an integration test than a unit test per se, but critical to ensure we don't introduce filtering performance issues as we do these big changes.